Model/ResourceModel/NodeType/Product.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Snowdog\Menu\Model\ResourceModel\NodeType;
+
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Store\Model\Store;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product as CoreProduct;
+use Magento\Framework\EntityManager\MetadataPool;
+
+class Product extends AbstractNode
+{
+    /**
+     * @var MetadataPool
+     */
+    private $metadataPool;
+
+    public function __construct(
+        ResourceConnection $resource,
+        MetadataPool $metadataPool
+    ) {
+        $this->metadataPool = $metadataPool;
+        parent::__construct($resource);
+    }
+
+    /**
+     * @return array
+     * @throws \Exception
+     */
+    public function fetchConfigData()
+    {
+        $metadata = $this->metadataPool->getMetadata(ProductInterface::class);
+        $identifierField = $metadata->getIdentifierField();
+        $connection = $this->getConnection('read');
+
+        $select = $connection->select()->from(
+            ['e' => $this->getTable('catalog_product_entity')],
+            [$identifierField, 'sku']
+        );
+
+        return $connection->fetchAll($select);
+    }
+
+    /**
+     * @param int   $storeId
+     * @param array $productIds
+     *
+     * @return array
+     */
+    public function fetchData($storeId = Store::DEFAULT_STORE_ID, $productIds = [])
+    {
+        $connection = $this->getConnection('read');
+        $table = $this->getTable('url_rewrite');
+        $select = $connection
+            ->select()
+            ->from($table, ['entity_id', 'request_path'])
+            ->where('entity_type = ?', 'product')
+            ->where('redirect_type = ?', 0)
+            ->where('store_id = ?', $storeId)
+            ->where('entity_id IN (' . implode(',', $productIds) . ')');
```

Comment:
As in the previous lines you use `?` to add values in where clause, I think we can use it here also so we can keep the same structure (it can be applied to the below methods also)